test/appCoinsTimelock.js

```diff
@@ -0,0 +1,132 @@
+
+        await AppCoinsTimelockInstance.allocateFunds(testAddresses[0], aditionalAmount);
+
+        const auxBalance = await AppCoinsTimelockInstance.getBalanceOf(testAddresses[0]);
+
+        expect(auxBalance.toNumber())
+        .to.be.equal(testAmounts[0] + aditionalAmount, "The contract is not saving the correct amount");
+
+    })
+
+    it('should store a bulk of addresses', async function () {
+        const releaseTime = (new Date()).getTime();
+
+        AppCoinsTimelockInstance = await AppCoinsTimelock.new(appcInstance.address, releaseTime);
+        TestUtils.setContractInstance(AppCoinsTimelockInstance);
+
+        await appcInstance.approve(AppCoinsTimelockInstance.address, testAmounts[0], { from: accounts[0] });
+
+        let totalamount = 0;
+
+        testAmounts.map((amount) => totalamount += amount);
+
+        await appcInstance.approve(AppCoinsTimelockInstance.address, totalamount, { from: accounts[0] });
+
+        await AppCoinsTimelockInstance.allocateFundsBulk(testAddresses, testAmounts);
+
+        const auxBalance =
+            await AppCoinsTimelockInstance.getBalanceOf(testAddresses[0]);
+
+        expect(auxBalance.toNumber())
+        .to.be.equal(testAmounts[0], "The contract is not saving the correct bulks amounts");
+    })
+
+    it('should enable release funds after time expires', async function () {
+        const releaseTime = (new Date("2000-01-01")).getTime();
+
+        AppCoinsTimelockInstance = await AppCoinsTimelock.new(appcInstance.address, releaseTime);
+        TestUtils.setContractInstance(AppCoinsTimelockInstance);
+
+        await appcInstance.approve(AppCoinsTimelockInstance.address, testAmounts[0], { from: accounts[0] });
+
+        await AppCoinsTimelockInstance.allocateFunds(testAddresses[0], testAmounts[0]);
+
+        await AppCoinsTimelockInstance.release(testAddresses[0], { from: accounts[0] });
+
+        const auxBalance = await AppCoinsTimelockInstance.getBalanceOf(testAddresses[0]);
+
+        expect(auxBalance.toNumber())
+        .to.be.equal(0, "The contract is releasing the correct amount");
+    })
+
+    it('should revert if the release time has not expired', async function () {
+        const releaseTime = (new Date("2099-01-01")).getTime();
+
+        AppCoinsTimelockInstance = await AppCoinsTimelock.new(appcInstance.address, releaseTime);
+        TestUtils.setContractInstance(AppCoinsTimelockInstance);
+
+        await appcInstance.approve(AppCoinsTimelockInstance.address, testAmounts[0], { from: accounts[0] });
+
+        await AppCoinsTimelockInstance.allocateFunds(testAddresses[0], testAmounts[0]);
+
+        await AppCoinsTimelockInstance.release(testAddresses[0], { from: accounts[0] }).catch(
+            (err) => {
+                reverted = expectRevert.test(err.message);
+            });
+        expect(reverted).to.be.equal(true,"Revert expected");
+    })
+
```

Comment:
I think an additional test should be created with a time travel feature of ganache, as @zepcp mentioned earlier.
From what i've researched you can base it on this:
https://medium.com/coinmonks/testing-time-dependent-logic-in-ethereum-smart-contracts-1b24845c7f72